src/main/java/ml/shifu/shifu/core/alg/TensorflowTrainer.java

```diff
@@ -0,0 +1,144 @@
+
+    private char delimiter;
+
+    private String lossFunc;
+
+    private String optimizer;
+    
+    private int epoch = 100;
+
+    public TensorflowTrainer(ModelConfig modelConfig, List<ColumnConfig> ccList) {
+        this.modelConfig = modelConfig;
+        this.ccList = ccList;
+        this.pathFinder = new PathFinder(modelConfig);
+        
+        for(int i = 0; i < ccList.size(); i++) {
+            ColumnConfig cc = ccList.get(i);
+            if(cc.isTarget()) {
+                targetColumnNum = i;
+            } else if(cc.isFinalSelect()) {
+                seletectedColumnNums.add(i);
+            }
+        }
+        if(seletectedColumnNums.size() == 0) {
+            for(int i = 0; i < ccList.size(); i++) {
+                ColumnConfig cc = ccList.get(i);
+                if(cc.isTarget()) {
+                    continue;
+                }
+                seletectedColumnNums.add(i);
+            }
+        }
+
+        ModelTrainConf modelTrainConf = modelConfig.getTrain();
+        learningRate = (double) modelTrainConf.getParams().get(CommonConstants.LEARNING_RATE);
+        actFuncs = (List<String>) modelTrainConf.getParams().get(CommonConstants.ACTIVATION_FUNC);
+        hiddenLayerNodes = (List<Integer>) modelTrainConf.getParams().get(CommonConstants.NUM_HIDDEN_NODES);
+        hiddenLayers = hiddenLayerNodes.size();
+        inputDataPath = pathFinder.getNormalizedDataPath();
+        alg = modelConfig.getAlgorithm();
+        delimiter = modelConfig.getDataSetDelimiter().charAt(0);
+        lossFunc = (String) modelTrainConf.getParams().get(CommonConstants.TF_LOSS);
+        optimizer = (String) modelTrainConf.getParams().get(CommonConstants.TF_OPTIMIZER);
+        epoch = modelTrainConf.getNumTrainEpochs();
+    }
+
+    public void train() throws IOException {
+        List<String> commands = buildCommands();
+        ProcessBuilder pb = new ProcessBuilder(commands);
+        pb.directory(new File("./"));
+        LOGGER.info("Start trainning sub process. Commands {}", commands.toString());
+        Process p = pb.start();
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<String> buildCommands() {
+        List<String> commands = new ArrayList<String>();
+
+        String actFuncStr = actFuncs.toString();
+        actFuncStr = actFuncs.toString().substring(1, actFuncStr.length() - 1);
+        String hiddenLayerNodesStr = hiddenLayerNodes.toString();
+        hiddenLayerNodesStr = hiddenLayerNodesStr.substring(1, hiddenLayerNodesStr.length() - 1);
+        String seletectedColumnNumsStr = seletectedColumnNums.toString();
+        seletectedColumnNumsStr = seletectedColumnNumsStr.substring(1, seletectedColumnNumsStr.length() - 1);
+        String delimiterStr  = String.valueOf(delimiter);
+        if((delimiter ^ '|') * (delimiter ^ '&') * (delimiter ^ '>') * (delimiter ^ '<') == 0) {
+            delimiterStr = "\\" + delimiter;
+        }
+
+        commands.add("python");
```

Comment:
python should be in PATH, in the future, maybe need set installed python bin in path

Reply:
I think we should discuss it later. A python runtime env maybe 100 times larger than shifu pakcage.